### PR TITLE
Raise File.Error exception on non existing file

### PR DESCRIPTION
**Pull request by gpopides**
refs #137

### CODE REVIEW

**Comment by mruoss** [comment]
Thanks for this. It does solve the issue at hand. However, File.read() also returns other tuples. So maybe there's other cases to deal with at some point. But this solves already the most common one.

Hmm... you should not have such a hard time with the pipeline. Let me fix that on develop branch so you can rebase.

**Comment by gpopides** [comment]
Hey, i thought about the other tuples but i was not sure if `File.Error` is the right exception for all of them.  Let me know if you want to address this in this PR

**Comment by mruoss** [comment]
You can rebase now and pipeline should be fine. 
As for your question, I'm undecided on how to deal with this. One possibility is to refactor so that `from_file/2` calls `from_file!/2`, rescues exceptions and turns them into tuples. But afaik that's generally not a very good idea because exceptions should be avoided... Maybe a topic for the elixir forum...

**Comment by gpopides** [comment]
`File.read!/1` returns always `File.Error`, so i think always raising `File.Error` from `from_file!/2` would be an option.

**Comment by mruoss** [comment]
What if in `render` where we know the error comes from `File.read` we turn it into a `{:error, %File.Error{...}}` tuple with `:reason` set accordingly?

Also see [File.read!](https://github.com/elixir-lang/elixir/blob/89e57882e120e77d755477b96ecfa15032dd582d/lib/elixir/lib/file.ex#L355)

**Comment by gpopides** [comment]
`from_file` also may return a yaml parsing error.  Do we want to distinguish handling between `File.Error` and `Yaml.ParsingError` ?

**Comment by mruoss** [comment]
If we turn the `{:error, atom()}` tuple in `render/2` into a `{:error, %File.Error{}}` tuple, we can leave the code in `from_file!` as is. It's just gonna [raise](https://github.com/coryodaniel/k8s/blob/develop/lib/k8s/resource.ex#L146) the `File.Error` (or `Yaml.ParsingError` if is returned):

File: lib/k8s/client/dynamic_http_provider.ex [hunk 1]
```diff
@@ -17,6 +17,18 @@ defmodule K8s.Client.DynamicHTTPProvider do
   defdelegate headers(method, request_options), to: K8s.Client.HTTPProvider
 
   @impl true
+  @spec handle_response(
```
Comment (mruoss) [inline lib/k8s/client/dynamic_http_provider.ex]: This is unrelated and should not be necessary now... right?

**Comment by mruoss** [comment]
Thanks @gpopides. Committed a slight change in 6990b8375952a69b7ef3fba48a35b58671e7b492.
